### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by maleck13**
moving client pkg out of mobile pkg. Some other general refactors and additional tests

### CODE REVIEW

**Comment by maleck13** [comment]
@philbrookes FYI moving things around a bit nothing major. Will merge once it passes

File: pkg/mobile/integration/service.go [hunk 1]
```diff
@@ -130,15 +117,16 @@ func (ms *MobileService) MountSecretForComponent(svcCruder mobile.ServiceCruder,
 	}
 
 	//find the clientService secret name
-	css, err := svcCruder.List(ms.filterServices([]string{clientService}))
-	if err != nil || len(css) == 0 {
+	cSeviceList, err := svcCruder.List(filterServices([]string{clientService}))
```
Comment (philbrookes) [inline pkg/mobile/integration/service.go]: typo here: `cServiceList`

File: pkg/mobile/types.go [hunk 2]
```diff
@@ -94,6 +94,9 @@ func (at AppTypes) String() string {
 //ValidAppTypes is a list of valid app types
 var ValidAppTypes = AppTypes{"cordova", "android", "iOS"}
 
+//ServiceTypes are the service types that we are aware of and support
+var ServiceTypes = []string{"fh-sync-server", "keycloak", "custom"}
```
Comment (philbrookes) [inline pkg/mobile/types.go]: I'm uncomfortable with this in general - not specific to this PR; I don't like that we have to recompile, rebuild and redeploy the API server anytime we want to add a new mobile service.

Comment (maleck13) [reply pkg/mobile/types.go]: yes agree, however I am chatting with the service catalog guys around having labels on the secrets https://github.com/kubernetes-incubator/service-catalog/issues/1222 which I think will help. Also can be moved out to an env var at somepoint

**Pull request merged**